PurchasesTests/SubscriberAttributes/SubscriberAttributeTests.swift

```diff
@@ -0,0 +1,93 @@
+        let subscriberAttribute = RCSubscriberAttribute(key: "a key",
+                                                        value: "a value",
+                                                        appUserID: "user id")
+        expect(subscriberAttribute.key) == "a key"
+        expect(subscriberAttribute.value) == "a value"
+        expect(subscriberAttribute.appUserID) == "user id"
+        expect(subscriberAttribute.setTime).to(beCloseTo(Date(), within: 0.5))
+        expect(subscriberAttribute.isSynced) == false
+
+        let setTime = Date()
+        let subscriberAttribute2 = RCSubscriberAttribute(key: "another key",
+                                                         value: "another value",
+                                                         appUserID: "user id2",
+                                                         isSynced: true,
+                                                         setTime: setTime)
+        expect(subscriberAttribute2.key) == "another key"
+        expect(subscriberAttribute2.value) == "another value"
+        expect(subscriberAttribute2.appUserID) == "user id2"
+        expect(subscriberAttribute2.setTime) == setTime
+        expect(subscriberAttribute2.isSynced) == true
+    }
+
+    func testInitWithDictionarySetsRightValues() {
+        let key = "some key"
+        let value = "some value"
+        let appUserID = "68asdfa4g3210"
+        let setTime = NSDate()
+        let isSynced = true
+        let subscriberDict: [String: NSObject] = [
+            "key": NSString(string: key),
+            "value": NSString(string: value),
+            "appUserID": NSString(string: appUserID),
+            "setTime": setTime,
+            "isSynced": NSNumber(booleanLiteral: isSynced),
+        ]
+
+        let subscriberAttribute = RCSubscriberAttribute(dictionary: subscriberDict)
+
+        expect(subscriberAttribute.key) == key
+        expect(subscriberAttribute.value) == value
+        expect(subscriberAttribute.appUserID) == appUserID
+        expect(subscriberAttribute.setTime as NSDate) == setTime
+        expect(subscriberAttribute.isSynced) == isSynced
+    }
+
+    func testAsDictionaryReturnsCorrectFormat() {
+        let key = "some key"
+        let value = "some value"
+        let appUserID = "68asdfa4g3210"
+        let subscriberAttribute = RCSubscriberAttribute(key: key,
+                                                        value: value,
+                                                        appUserID: appUserID)
+
+        let receivedDictionary = subscriberAttribute.asDictionary()
+        expect(receivedDictionary.keys.count) == 5
+
+        expect(receivedDictionary["key"] as? String) == key
+        expect(receivedDictionary["value"] as? String) == value
+        expect(receivedDictionary["appUserID"] as? String) == appUserID
+        expect(receivedDictionary["setTime"] as! Date).to(beCloseTo(Date(), within: 0.5))
```

Comment:
I wasn't sure whether it was worth the trouble doing the DateProvider / stubbedNow trick to stub the date, this instead just checks that the date is very, very close to now. 
There are downsides, of course, because it's technically possible (albeit highly unlikely) that this lines executes long enough after the creation of the object that this fails. 
@vegaro thoughts? Should I just go ahead and add either DateProvider or stubbedNow?

Reply:
I added a DateProvider in the Android implementation to check against this. If you prefer the stubbedNow go ahead too